pkg/innerring/innerring.go

```diff
@@ -256,18 +239,56 @@ func (s *Server) Start(ctx context.Context, intError chan<- error) (err error) {
 	go s.fsChainListener.ListenWithError(ctx, fsChainErr)  // listen for neo:fs events
 	go s.mainnetListener.ListenWithError(ctx, mainnnetErr) // listen for neo:mainnet events
 
-	if err = s.epochTimer.Reset(); err != nil {
-		return fmt.Errorf("could not start new epoch block timer: %w", err)
-	}
-	if err = initialEpochTicker.Reset(); err != nil {
-		return fmt.Errorf("could not start initial new epoch block timer: %w", err)
+	if err = s.resetEpochTimer(false); err != nil {
+		return fmt.Errorf("could not reset epoch timer: %w", err)
 	}
 
 	s.startWorkers(ctx)
 
 	return nil
 }
 
+func (s *Server) resetEpochTimer(afterNewEpochEvent bool) error {
+	var nextTickAfter time.Duration
+	if afterNewEpochEvent {
+		// get current epoch duration
+		epochDuration, err := s.netmapClient.EpochDuration()
+		if err != nil {
+			return fmt.Errorf("can't read epoch duration: %w", err)
+		}
+
+		lastTick, err := s.netmapClient.LastEpochBlock()
+		if err != nil {
+			return fmt.Errorf("can't read last epoch block: %w", err)
+		}
+		lastTickH, err := s.fsChainClient.GetBlockHeader(lastTick)
+		if err != nil {
+			return fmt.Errorf("can't read last tick's block header (#%d): %w", lastTick, err)
+		}
+
+		blockHeight, err := s.fsChainClient.BlockCount()
+		if err != nil {
+			return fmt.Errorf("can't get FS chain height: %w", err)
+		}
+		currentH, err := s.fsChainClient.GetBlockHeader(blockHeight - 1)
+		if err != nil {
+			return fmt.Errorf("can't read last current block header (#%d): %w", blockHeight, err)
+		}
+
+		nextTickAfter = epochTickAfter(lastTickH.Timestamp, currentH.Timestamp, epochDuration)
+	}
+
+	if nextTickAfter == 0 {
+		for _, h := range s.newEpochTickHandlers() {
+			h()
```

Comment:
Seems to be wrong, consider delayed epoch tick, it's not correct to call handlers in this case (even though it shouldn't hurt at `Start()`).

Reply:
we can discuss it. in fact, these were the last lines i added; it does not affect timers, it is more about #3397 situation. it is likely the best thing an ir that has not received an epoch event but that truly believes that it should have ticked it already can do

Reply:
You can differentiate missing tick from node missing a tick.